### PR TITLE
Issue#402

### PR DESCRIPTION
**Pull request by Cijin**
Removes the optional chaining syntax '?' from the AppNav.tsx file.

This syntax is causing fresh installs to fail on some node versions.

### CODE REVIEW

File: components/AppNav.tsx [hunk 1]
```diff
@@ -68,7 +68,8 @@ const AuthLink: React.FC<AuthLinkProps> = ({ session }) => {
 const AuthButton: React.FC<AuthButtonProps> = ({ initial, username }) => {
   const [logoutUser, { data }] = useLogoutMutation()
   useEffect(() => {
-    if (data?.logout?.success) {
+    const success = _.get(data, 'logout', false)
```
Comment (songz) [inline components/AppNav.tsx:71]: shouldn't it be `_.get(data, 'logout.success', false)`

File: pages/curriculum.tsx [hunk 2]
```diff
@@ -69,7 +71,7 @@ export const Curriculum: React.FC<GetAppProps> = ({ data }) => {
         currentState={lessonState}
         reviewUrl={`/review/${id}`}
         challengesUrl={`/curriculum/${id}`}
-        docUrl={lesson?.docUrl || ''}
+        docUrl={lesson.docUrl || ''}
```
Comment (hwong0305) [inline pages/curriculum.tsx:74]: Shouldn't this be `_.get(lesson, 'docUrl', '')` or does the GraphQL definition already guarantee that it exists?

Comment (Cijin) [reply pages/curriculum.tsx:74]: I thought since it's got a fallback `|| ''`. Wasn't completely sure about it.

Comment (anthonykhoa) [reply pages/curriculum.tsx:74]: I believe @hwong0305 is right. If `lesson` is undefined, then `lesson.docUrl` would cause a TypeError

Comment (Cijin) [reply pages/curriculum.tsx:74]: Using `_.get(lesson, 'docUrl', '')` creates a type error saying `type Maybe<String> is not assignable to type string`. How can I fix that?
If lesson is undefined an error would be thrown earlier saying `<h1>Bad Data</h1>`

File: pages/curriculum.tsx [hunk 3]
```diff
@@ -48,11 +49,12 @@ export const Curriculum: React.FC<GetAppProps> = ({ data }) => {
   const lessonsToRender: React.ReactElement[] = lessons.map((lesson, idx) => {
     const id = _.get(lesson, 'id', idx) as number
     const status = lessonStatusMap[id]
+    const passed = _.get(status, 'isPassed', false)
     let lessonState = ''
     if (idx === lessonInProgressIdx) {
       lessonState = 'inProgress'
     }
-    if (status?.isPassed) {
```
Comment (songz) [inline pages/curriculum.tsx:55]: put variables declarations close to where it is used. so move `const passed = ....` to here.

File: pages/curriculum.tsx (hunk 2 above)
Comment (anthonykhoa) [reply pages/curriculum.tsx:74]: Try `_.get(lesson, 'docUrl', '') as string || ''`

**Review by songz — APPROVED** [review]

**Review by ggwadera — APPROVED** [review]

**Review by anthonykhoa — APPROVED** [review]

**Pull request merged**